internal/db/repositories/user_repository.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/TheDevExperiment/server/internal/log"
+	"github.com/TheDevExperiment/server/internal/utility/funnynamer"
+	"github.com/TheDevExperiment/server/internal/utility/jwt"
+)
+
+type UserRepository struct {
+	collection *mongo.Collection
+}
+type UserModel struct {
+	Id              primitive.ObjectID `bson:"_id,omitempty" `
+	IsGuest         bool               `bson:"isGuest,omitempty"`
+	GuestAuthSecret string             `bson:"guestAuthSecret,omitempty"`
+	CreatedAt       primitive.DateTime `bson:"createdAt,omitempty"`
+	LastModified    primitive.DateTime `bson:"lastModified,omitempty"`
+	DisplayName     string             `bson:"displayName,omitempty"`
+	CountryID       string             `bson:"countryId,omitempty"`
+	CityID          string             `bson:"cityId,omitempty"`
+	Age             string             `bson:"age,omitempty"`
+	Score           _score             `bson:"score,omitempty"`
+	IsActive        bool               `bson:"isActive,omitempty"`
+	DeletionReason  string             `bson:"deletionReason,omitempty"`
+}
+type UserUpdateModel struct {
+	// Id              primitive.ObjectID `bson:"_id,omitempty" `
+	IsGuest         bool               `bson:"isGuest,omitempty"`
+	GuestAuthSecret string             `bson:"guestAuthSecret,omitempty"`
+	LastModified    primitive.DateTime `bson:"lastModified,omitempty"`
+	DisplayName     string             `bson:"displayName,omitempty"`
+	// CountryID       string             `bson:"countryId,omitempty"`
+	// CityID          string             `bson:"cityId,omitempty"`
+	// Age            string `bson:"age,omitempty"`
+	Score          _score `bson:"score,omitempty"`
+	IsActive       bool   `bson:"isActive,omitempty"`
+	DeletionReason string `bson:"deletionReason,omitempty"`
+}
+
+type _score struct {
+	RatingScoreSum int `bson:"ratingScoreSum"`
+	RatingCount    int `bson:"ratingCount"`
+}
+
+func NewUserRepository() *UserRepository {
+	return &UserRepository{db.GetCollection(viper.GetString("mongodb.db_name"), "users")}
+}
+
+func (r *UserRepository) Find(ctx context.Context, filter interface{}) ([]UserModel, error) {
+	cursor, err := r.collection.Find(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	var results []UserModel
+	err = cursor.All(ctx, &results)
+	if err != nil {
+		return nil, err
+	}
+	return results, nil
+}
+func (r *UserRepository) Create(ctx context.Context, userAge string, countryId string, cityId string) (UserModel, error) {
+	doc := UserModel{
+		IsGuest:         true,
+		GuestAuthSecret: "",
+		CreatedAt:       primitive.NewDateTimeFromTime(time.Now()),
+		LastModified:    primitive.NewDateTimeFromTime(time.Now()),
+		DisplayName:     funnynamer.GetFunnyName(),
+		CountryID:       countryId,
+		CityID:          cityId,
+		Age:             userAge,
+		Score:           _score{RatingScoreSum: 0, RatingCount: 0},
+		IsActive:        true,
+		DeletionReason:  "",
+	}
+
+	result, err := r.collection.InsertOne(context.Background(), doc)
```

Comment:
use the ctx that is passed.
why? right now its not supported but in future I will add support for context cancellations.. which means, if a user disconnects our client, we will also drop the req on our end. Ctx has this info